src/Kestrel.Transport.Sockets/Internal/SocketTrace.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal
+{
+    public class SocketTrace : ISocketTrace
+    {
+        // ConnectionRead: Reserved: 3
+
+        private static readonly Action<ILogger, string, Exception> _connectionPause =
+            LoggerMessage.Define<string>(LogLevel.Debug, 4, @"Connection id ""{ConnectionId}"" paused.");
```

Comment:
Won't these IDs clash with other IDs? What's the scope of an event ID?

Reply:
The IDs match mostly to make the cross-transport tests easier. It's in the "Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets" scope, so there shouldn't be any conflicts.